verticapy/learn/vmodel.py

```diff
@@ -4182,14 +4224,20 @@ def fit(self, input_relation: Union[str, vDataFrame], X: list = []):
         query += ", ".join(
             ["{} = {}".format(elem, parameters[elem]) for elem in parameters]
         )
+        if self.type == "BisectingKMeans" and isinstance(
+            verticapy.options["random_state"], int
+        ):
+            query += ", kmeans_seed={}, id_column='{}'".format(
+                verticapy.options["random_state"], id_column_name
+            )
         query += ")"
         try:
             executeSQL(query, "Fitting the model.")
-            if isinstance(input_relation, vDataFrame):
-                drop_if_exists(relation, method="view")
+            if tmp_view:
+                drop(relation, method="view")
         except:
-            if isinstance(input_relation, vDataFrame):
-                drop_if_exists(relation, method="view")
+            if tmp_view:
+                drop(relation, method="view")
```

Comment:
I don't understand the logic of this try-except.

Reply:
the except is in case it did not work, so I have to drop the tmp view, is there a better way to do it?